src/main/java/org/mastodon/mamut/treesimilarity/ZhangUnorderedTreeEditDistance.java

```diff
@@ -136,10 +148,25 @@ public class ZhangUnorderedTreeEditDistance< T >
 	private final Map< Pair< Tree< T >, Tree< T > >, Double > attributeDistances;
 
 	/**
-	 * Simple default cost function that returns the absolute value of the difference between two costs,
-	 * if both costs exist or the cost value of the other, if one value is {@code null}.
+	 * Cost function as used in the treex library tests that returns the absolute value of the difference between two attributes,
```

Comment:
The `ZhangUnorderedTreeEditDiastance` is on itself a very long java class, with a lot of complexity. This PR adds more complexity. I would strongly recommend that the class only implements the algorithm described in the paper Zhang.

The methods `normalizedDistance`, `averageDistance`, `guinardDistance` that implement somehow related tree distance should be moved to a different class that is maybe called `TreeDistances`. Also the cost functions `TREE_X_COST_FUNCTION`, `GUIGNARD_COST_FUNCTION` should be moved to this new class.

Reply:
Good advice. I introduced a new class TreeDistances.
I moved the distance methods to this class.